server/gokbg3/grails-app/controllers/org/gokb/rest/CuratoryGroupsController.groovy

```diff
@@ -271,30 +271,43 @@ class CuratoryGroupsController {
     def result = ['result':'ERROR',
                   'params': params]
     if (!params.superordinateId || !params.subordinateId){
-      response.setStatus(422)
+      response.status = 422
       result.message = "Missing params. Requested parameters are 'superordinateId' and 'subordinateId'"
     }
-    else{
+    else if (params.superordinateId == params.subordinateId) {
+      response.status = 422
+      result.message = "A group cannot be set as its own superordinate."
+    }
+    else {
       CuratoryGroup superordinate = CuratoryGroup.get(genericOIDService.oidToId(params.superordinateId))
       CuratoryGroup subordinate = CuratoryGroup.get(genericOIDService.oidToId(params.subordinateId))
-      if (!superordinate || !subordinate){
-        response.setStatus(404)
+      if (!superordinate || !subordinate) {
+        response.status = 404
         result.message = "CuratoryGroup combination not found for superordinate id ${params.superordinateId} and subordinate id ${params.subordinateId}."
       }
-      else if (!(superordinate.type.level > subordinate.type.level)){
-        response.setStatus(409)
+      else if (!subordinate.type || !superordinate.type) {
+        response.status = 400
+        result.message = "One or both of the provided groups do not have any type set."
+      }
+      else if (superordinate.type.level.levelNumber == 75) {
```

Comment:
`CuratoryGroupType.Level.CENTRAL.value` is more typing than `75` but still preferable.